allennlp/training/trainer.py

```diff
@@ -980,11 +980,34 @@ def from_params(cls,
         grad_norm = params.pop_float("grad_norm", None)
         grad_clipping = params.pop_float("grad_clipping", None)
         lr_scheduler_params = params.pop("learning_rate_scheduler", None)
+        fine_tuning = params.pop("fine-tuning", None)
+        optimizer_params = params.pop("optimizer")
+        if fine_tuning == "discriminative" or \
+                (lr_scheduler_params is not None and "freeze" in lr_scheduler_params):
+            if "parameter_groups" in optimizer_params:
+                modules = [group[0] for group in optimizer_params["parameter_groups"]]
+            else:
+                # split up parameters in groups based on modules
+                modules = [[m] for m in model._modules]
+            if fine_tuning == "discriminative":
+                # use a different learning rate for every parameter group
+                assert "lr" in optimizer_params, \
+                    "Error: A base learning rate is required for " \
+                    "discriminative fine-tuning."
+                lr = optimizer_params["lr"]
+                decay_factor = 0.38
```

Comment:
Let's pull this out into a configurable parameter in `optimizer_params` with this as the default as deeper / shallower models may need a different decay factor.